src/components/main-page/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,62 @@
+import { useTranslation } from "next-i18next";
+import { Section, Heading, Text, ScheduleItem } from "components/ui";
+import { iconsSet } from "./../../ui/ScheduleItem/ScheduleItem";
+import { scheduleList } from "./config";
+import styles from "./Schedule.module.scss";
+
+type Agenda = {
+  id: number;
+  title: string;
+  icon?: string;
+  start: string;
+  end: string;
+};
+
+export function Schedule() {
+  const { t } = useTranslation();
+
+  const days = t("schedule.days", {
+    returnObjects: true,
+  }) as { name: string; agenda: Agenda[] }[];
+
+  function createScheduleElement(item: Agenda): JSX.Element {
+    const timer = scheduleList.find((el) => el.id === item.id);
+
+    let color: "blue" | "orange" | "lightBlue" = "blue";
+
+    if (timer) {
+      color =
+        timer?.endTimestamp.getTime() < new Date().getTime()
+          ? "lightBlue"
+          : timer?.startTimestamp?.getTime() < new Date().getTime()
+          ? "orange"
+          : "blue";
+    }
+
+    return (
+      <ScheduleItem
+        title={item.title}
+        icon={item.icon as keyof typeof iconsSet}
```

Comment:
This typing rule is declared in `ScheduleItem.tsx`, maybe it is not necessary, but it could help  😅 I will use your second suggestion and export `IconType` and put it here instead of `keyof typeof iconsSet` to improve code readability.